Slim/Player/Protocols/HTTP.pm

```diff
@@ -286,9 +339,59 @@ sub canDirectStream {
 	return $url;
 }
 
+# TODO: what happens if this method is overloaded in a sub-class that does not call its parents method
+sub canDirectStreamSong {
+	my ( $class, $client, $song ) = @_;
+	
+	# can't go direct if we are synced or proxy is set by user
+	my $direct = $class->canDirectStream( $client, $song->streamUrl(), $class->getFormatForURL() );
+	return 0 unless $direct;
+	
+	# no header or stripHeader flag has precedence
+	return $direct if $song->stripHeader || !defined $song->track->initial_block_type;
+	
+	# with dynamic header 2, always go direct otherwise only when not seeking
+	if ($song->track->initial_block_type == Slim::Schema::RemoteTrack::INITIAL_BLOCK_ALWAYS || $song->seekdata || $song->track->audio_initiate) {
```

Comment:
Same here, we should not try to test audio_initiate which is not supposed to exist